larana/OpticalDetector/OpHitFinder/AlgoSlidingWindow.cxx

```diff
@@ -154,7 +154,8 @@ namespace pmtana {
         // If there's a pulse, end we where in in_post, end the previous pulse first
         if (in_post) {
           // Find were
-          _pulse.t_end = i - buffer_num_index - 1;
+          _pulse.t_end = static_cast<int>(i) - buffer_num_index;
+          if (_pulse.t_end > 0) --_pulse.t_end; // leave a gap, if we can
```

Comment:
These are always awkward problems.  Perhaps you might like this solution:

```.cpp
auto const diff = static_cast<int>(i) - buffer_num_index;
_pulse.t_end = diff - static_cast<int>(diff > 0); // leave a gap, if we can  
```

Reply:
It's elegant, but I suspect it's harder to interpret by the casual reader.

Reply:
I agree.  Keep as is.